lib/matplotlib/axes/_axes.py

```diff
@@ -2665,18 +2670,24 @@ def sign(x):
                 xy = endpt, yc
 
             if orientation == "vertical":
-                xytext = 0, sign(dat) * padding
+                xytext = 0, yinverted * sign(dat) * padding
```

Comment:
```suggestion
                ydirection = -1 if yinverted else 1
                xytext = 0, ydirection * sign(dat) * padding
```
Same for `xinverted` below